gossip/discovery/discovery_impl.go

```diff
@@ -339,8 +339,8 @@ func (d *gossipDiscoveryImpl) handleMsgFromComm(msg protoext.ReceivedMessage) {
 		}
 
 		var internalEndpoint string
-		if m.Envelope.SecretEnvelope != nil {
-			internalEndpoint = protoext.InternalEndpoint(m.Envelope.SecretEnvelope)
+		if memReq.SelfInformation.SecretEnvelope != nil {
+			internalEndpoint = protoext.InternalEndpoint(memReq.SelfInformation.SecretEnvelope)
```

Comment:
Tests?

> While this has no semantic effect, because the consumer of the internal endpoint doesn't make use of the field, the field might still be logged, and we should use the correct envelope which contains the internal endpoint and not the incorrect envelope.

That paragraph (and the fact that this PR is backported two releases) implies this is something that should be validated via automated test and assertions.

Reply:
Tests for what? we don't use this field in production code. At most it can be logged but nothing more.

Reply:
Added an assertion